iterator_practice.cpp

```diff
@@ -1,16 +1,13 @@
 #include <iostream>
-#include <iterator> // for iterator
 #include <vector> // for vectors 
 
-using namespace std;
-
-int main()
-{
-    vector<int> ar = {2,4,6,8,10};
-    vector<int> ::iterator ptr;
-    cout << "My vector elements are: ";
-    for (ptr = ar.begin(); ptr < ar.end (); ptr++)
-        cout << *ptr<< " ";
-
-    return 0; 
-}
\ No newline at end of file
+int main() {
+  const std::vector<int> evens{2,4,6,8,10};
+  std::cout << "My vector elements are: ";
+  for (auto it = evens.begin(); it < evens.end(); ++it) {
```

Comment:
We usually name iterators `it` if there's only one at a time around. `auto` let's you not worry about what the iterator's type is, since it's kind of self-explanatory here. You also don't need to declare the iterator until you use it. So it's better to declare it inside the for loop, since it's only used inside the loop. Once the loop is done, as I write it the iterator is disposed of, but in your version, it hangs around.